compiler/luci-interpreter/src/kernels/Concatenation.cpp

```diff
@@ -69,11 +69,32 @@ void Concatenation::configure()
   Shape output_shape = t0->shape();
   output_shape.dim(axis) = sum_axis;
 
-  // TODO S8 type needs more checking: quantization parameters of all input tensors and the output
-  //  tensor should be the same. Note that there is no such requirement for U8 type.
-  if (t0->element_type() == DataType::S8)
-    throw std::runtime_error("Unsupported type.");
-
+  // If input tensors are INT8 type then quantization parameters of all input tensors and the output
+  // should be the same
+  for (int32_t i = 0; i < num_inputs; ++i)
+  {
+    const Tensor *current_tensor = _inputs.at(i);
+    if (current_tensor->element_type() == DataType::S8)
+    {
+      LUCI_INTERPRETER_CHECK(current_tensor->quantized_dimension() ==
+                             output()->quantized_dimension());
+      const auto quantized_dimension = current_tensor->quantized_dimension();
+
+      LUCI_INTERPRETER_CHECK(current_tensor->zero_points().at(quantized_dimension) ==
+                             output()->zero_points().at(quantized_dimension));
+      LUCI_INTERPRETER_CHECK(current_tensor->scales().at(quantized_dimension) ==
+                             output()->scales().at(quantized_dimension));
```

Comment:
There are more then 1 element in zero points and scales if tensor is quantized "channel-wise".

For example, we could have tensor with dimensions `(1, 32, 32, 16)` and `quantized_dimension == 3`.
This means that `zero_points().size() == scales().size() == 16`.

So, to check correctness of quantization you should do something like this:
```suggestion
      auto size = current_tensor->zero_points().size();
      LUCI_INTERPRETER_CHECK(size == current_tensor->scales().size());
      LUCI_INTERPRETER_CHECK(size == output()->zero_points().size());
      LUCI_INTERPRETER_CHECK(size == output()->scales().size());
      for (size_t i = 0; i < rank; ++i)
      {
        LUCI_INTERPRETER_CHECK(current_tensor->zero_points()[i] == output()->zero_points()[i]);
        LUCI_INTERPRETER_CHECK(current_tensor->scales()[i] == output()->scales()[i]);
      }
```
P.s. check if this code is ok in ide and run formatter, I edited it in comment

Reply:
Done